plugin-core/src/main/java/appland/index/BasicAppMapMetadataExternalizer.java

```diff
@@ -10,13 +10,17 @@
 import java.io.IOException;
 
 class BasicAppMapMetadataExternalizer implements DataExternalizer<BasicAppMapMetadata> {
-    public static final BasicAppMapMetadataExternalizer INSTANCE = new BasicAppMapMetadataExternalizer();
+    static final @NotNull BasicAppMapMetadataExternalizer INSTANCE = new BasicAppMapMetadataExternalizer();
+
+    private BasicAppMapMetadataExternalizer() {
+    }
 
     @Override
     public BasicAppMapMetadata read(@NotNull DataInput in) throws IOException {
-        var name = IOUtil.readUTF(in);
-        var hasTestStatus = in.readBoolean();
+        var hasName = in.readBoolean();
+        var name = hasName ? IOUtil.readUTF(in) : null;
 
+        var hasTestStatus = in.readBoolean();
```

Comment:
Correct me if I'm wrong, but aren't `hasName` and `hasTestStatus` always going to be equal to each other because they are both the return value from `in.readBoolean()`? That would mean that `name` and `jsonId` would always be equal to each other, too. Is that intentional?

Reply:
ˋinˋ represents a stream of data (bytes), and ˋreadBooleanˋ has side-effects, because it reads from the stream and advances the current position. Calling it twice is reading to different values.